server/sys.go

```diff
@@ -0,0 +1,17 @@
+package server
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func HandleSignals(ctx context.Context) context.Context {
+	newCtx, stop := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
```

Comment:
do you recommend also listening to `SIGINT` for graceful shutdowns? its basically when we do `ctrl + c `. 
Ref: https://dsa.cs.tsinghua.edu.cn/oj/static/unix_signal.html